### ISSUE TITLE
Hub mains power logs a second 'unplugged' before power actually returns

### ISSUE THREAD

**Issue by aavdberg** [bvis/aegis-hass#386]
### Describe the bug

During a power outage, the hub mains-power state is logged as `unplugged` twice, while the mains power was still off the second time.

I expected one transition to `unplugged` when power was lost, and then no second change until power actually returned.

### Integration version:

[v1.15.1-beta.12](https://github.com/bvis/aegis-hass/releases/tag/v1.15.1-beta.12)

First transition to unplugged:

```text
2026-08-04 10:10:33.887 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub 002A1D5E: parsed 55 keys from STATUS_BODY
2026-08-04 10:10:33.887 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub 002A1D5E: externally_powered None -> False via STATUS_BODY (KEY_HUB_POWERED raw=00, keys=[...])
```

Later, while power was still not back yet, the hub kept logging refreshes from the same power-delta path:

```text
2026-08-04 10:16:10.081 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub 002A1D5E: STATUS_UPDATE push (#123): 30B9BA58=[0x03=03,0xc3=00]
2026-08-04 10:16:10.082 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub 002A1D5E: requesting fresh HTS snapshot after untrusted power delta
2026-08-04 10:16:10.083 DEBUG [custom_components.aegis_ajax.api.hts.client] Sent REQUEST_FULL_SETTINGS to 002A1D5E
2026-08-04 10:16:10.083 DEBUG [custom_components.aegis_ajax.api.hts.client] Sent REQUEST_FULL_STATUS to 002A1D5E
```

The same pattern repeats later:

```text
2026-08-04 10:43:34.960 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub 002A1D5E: STATUS_UPDATE push (#123): 30DCC7D6=[0x03=03,0xc3=00]
2026-08-04 10:43:34.961 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub 002A1D5E: requesting fresh HTS snapshot after untrusted power delta
```

### Expected behavior

The hub should only be marked `unplugged` once when mains power actually drops, and should not appear to flip again until `externally_powered` truly becomes `True`.

### Additional context

This looks related to the HTS power parsing / refresh path rather than the FCM push path.

**Comment by aavdberg** [comment]
<img width="574" height="420" alt="Image" src="https://github.com/user-attachments/assets/5f645d17-5397-4e4b-9913-5af474f28d2d" />

<img width="1178" height="966" alt="Image" src="https://github.com/user-attachments/assets/3b29ce7d-5b80-4a5e-bfcb-ca36d0046b8d" />

**Comment by bvis** [comment]
Thanks for the log excerpt — it was enough to find a real bug, and it is not the one the title describes. Both things need saying, so let me separate them.

**What your log proves.** These two lines are the same hub message taking two code paths at once:

```
10:16:10.081  STATUS_UPDATE push (#123): 30B9BA58=[0x03=03,0xc3=00]
10:16:10.082  requesting fresh HTS snapshot after untrusted power delta
```

The first line is a status update from **one of your devices** (`30B9BA58`), whose first value is `0x03` — the operational-state byte that most Ajax devices report. The integration also reads `0x03` as the hub's mains-power key, and the guard meant to tell those two apart only recognises one message layout. Yours uses a different one, so a routine per-device update is being misread as "the hub says mains power came back". Because that disagrees with the stored state, it requests a **full hub snapshot** every time — roughly 8.6 KB, repeating every ~27 minutes, while your hub is running on battery. That is the worst possible moment for it, so it is worth fixing on its own. A fix is going into the next beta.

**What it does not yet prove.** It does not explain a *second* `unplugged`. That misread requests a snapshot; it does not itself write the power state. For Home Assistant to log `unplugged` twice, the sensor has to have left that state in between — either to "plugged in", or to `unavailable` (a restart does this). Your log excerpt stops before whichever it was.

The integration already logs the line that settles it, at debug level, on every single change to that flag:

```
Hub 002A1D5E: externally_powered <old> -> <new> via <source> (KEY_HUB_POWERED raw=..., keys=[...])
```

You pasted the first one (`None -> False`). If you still have that log, the **next** one after it is the whole answer:

1. `False -> True via STATUS_BODY` — the hub genuinely reported mains power while it was still out, which is a different and more interesting problem.
2. `None -> False` again — the connection restarted and lost its memory of the previous state.
3. No second line at all — the flag never changed, the two `unplugged` entries are Home Assistant restarting around your power loss, and there is nothing to fix in that half.

One more thing worth asking, given the timing: you mentioned separately that your UPS ran out and you lost grid power. **Did Home Assistant itself restart between the two `unplugged` entries?** If it did, that alone explains option 3 and saves us both the log hunt.

**Comment by aavdberg** [comment]
Update on the two logs you asked about:

- In `home-assistant_aegis_ajax_2026-08-04T07-45-20.215Z.log` I only found one explicit power write:
  ```text
  2026-08-04 10:10:33.887 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub 002A1D5E: externally_powered None -> False via STATUS_BODY (KEY_HUB_POWERED raw=00, ...)
  ```
  After that, I do **not** see a `False -> True` transition in that capture, only repeated `requesting fresh HTS snapshot after untrusted power delta` lines.

- In `home-assistant_2026-08-04T13-51-26.793Z.log` I **do** see a genuine return to mains power:
  ```text
  2026-08-04 16:41:36.679 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub 002A1D5E: externally_powered None -> True via STATUS_BODY (KEY_HUB_POWERED raw=01, ...)
  ```
  and HA startup lines right after:
  ```text
  2026-08-04 16:41:46.656 INFO [homeassistant.bootstrap] Home Assistant initialized in 18.43s
  2026-08-04 16:41:46.656 INFO [homeassistant.core] Starting Home Assistant 2026.7.4
  ```

So for the first capture, I still don’t see evidence of a real `plugged in` change between the two `unplugged` entries. In the second capture, the hub does report mains power back on, but HA is also restarting around that time, so that log doesn’t answer the earlier duplicate-`unplugged` question cleanly.

**Comment by aavdberg** [comment]
Update from the log:

- `2026-08-04 09:14:58.139` — `requesting fresh HTS snapshot after untrusted power delta`
- `2026-08-04 09:14:58.231` — `externally_powered True -> False via STATUS_BODY ...`
- `2026-08-04 09:15:13.187` — `STATUS_UPDATE push (#123): 30D93A33=[0x02=1d,0xc3=00]`

So what I see is: first a real power delta to `False`, and then a device push/status update. In this fragment I do not see a second real power transition that would explain a second unplugged message; it looks more like a refresh/snapshot after the first delta.

**Comment by bvis** [comment]
**[`1.16.0-beta.2`](https://github.com/bvis/aegis-hass/releases/tag/v1.16.0-beta.2)** is out with the fix for the snapshot storm (#389).

Your hub will no longer request a full refresh every time one of your devices reports its status. A genuine mains-power change still asks for immediate confirmation, so nothing became slower to react — there is a test guarding exactly that, because narrowing this too far would have reintroduced #323.

HACS → the integration → three-dot menu → Redownload → tick "Show beta versions" → pick `1.16.0-beta.2` → restart Home Assistant. It also carries the logbook fix for #387.

**To be clear about what this does and does not settle:** it fixes the wasteful refreshes your log showed. It does not explain a *second* `unplugged` — that needs the sensor to have left the state in between, and nothing in the fix touches how the state is written. So the questions from my previous comment still stand, and the simplest one is likely to be decisive:

- **Did Home Assistant itself restart between the two `unplugged` entries?** You mentioned separately that your UPS ran out during this. If it restarted, that alone explains the second entry and there is nothing further to fix.
- If it did not, the next `externally_powered <old> -> <new> via <source>` line after the one you pasted is the answer.

No rush on either — take the outage recovery first.

**Comment by aavdberg** [comment]
@bvis the outage is normal in Ukraine 😁 this summer it was better but now with the hot weather it's getting worse again. I the winter it was terrible whole days with out power from the grid.

**Comment by bvis** [comment]
Your fragment settles it, and your own reading of it was right. Closing — with both halves explained separately, because they turned out to be two unrelated things.

**The second `unplugged` was a Home Assistant restart, not a flapping sensor.** Your log shows exactly one real transition, `True -> False`, and no second one — which is what you concluded too. The detail that confirms it is in the line you pasted first: it reads `externally_powered None -> False`, and that `None` means the integration had *no previous knowledge* of the hub at that moment. It only reads `None` on a fresh start. So the sensor never left `unplugged` and never flipped back; Home Assistant restarted when the UPS ran out, the entity went `unavailable` and then returned to `unplugged`, and the history renders that as a second entry. Nothing in the integration was misreporting your power state.

**The refresh storm was real and is fixed** in [`1.16.0-beta.2`](https://github.com/bvis/aegis-hass/releases/tag/v1.16.0-beta.2). That is the genuine bug your log caught: a routine status update from one of your devices was being counted as "the hub says mains power is back", and each one requested a full ~8.6 KB snapshot — repeating every few minutes while your hub was on battery, which is the worst possible time for it. That one was worth the report on its own.

If you still see `requesting fresh HTS snapshot after untrusted power delta` repeating during an outage on beta.2 or later, reopen this and I will take another look — that line is the exact tell.

Thanks for the log. Both halves came out of it, and the one that mattered was not the one in the title.

Given the outages are a routine fact of life there rather than a one-off, one thing worth saying: the hub reporting `unplugged` while running on battery is correct behaviour, and if the repeated `unavailable` gaps around restarts are a nuisance for automations, `not_from: unavailable` on state triggers is the guard for it — the bundled blueprints do this for the same reason.

**Comment by bvis** [comment]
Shipped in `1.16.0` stable: https://github.com/bvis/aegis-hass/releases/tag/v1.16.0. Thanks @aavdberg for the outage log that separated the real snapshot storm from the restart artifact.
